detox/src/Detox.js

```diff
@@ -144,33 +145,38 @@ class Detox {
     this._client.setNonresponsivenessListener(this._onNonresnponsivenessEvent.bind(this));
     await this._client.connect();
 
+    const invocationManager = new InvocationManager(this._client);
     const deviceDriver = driverRegistry.resolve(this._deviceConfig.type, {
       client: this._client,
+      invocationManager,
       emitter: this._eventEmitter,
     });
 
-    Object.assign(this, deviceDriver.matchers);
-
     this.device = new Device({
       deviceConfig: this._deviceConfig,
       emitter: this._eventEmitter,
       deviceDriver,
       sessionConfig,
     });
 
-    if (behaviorConfig.exposeGlobals) {
-      Object.assign(Detox.global, {
-        ...deviceDriver.matchers,
-        device: this.device,
-      });
-    }
-
     this._artifactsManager = new ArtifactsManager(this._artifactsConfig);
     this._artifactsManager.subscribeToDeviceEvents(this._eventEmitter);
     this._artifactsManager.registerArtifactPlugins(deviceDriver.declareArtifactPlugins());
 
     await this.device.prepare();
 
+    const matchers = matchersRegistry.resolve(this.device, {
+      invocationManager,
```

Comment:
@d4vidi, so, your plan is to inject `device` or a similar handle there, to ensure that the matchers will be able to pull the files from the device, after the screenshot is done?

Reply:
Bingo! The restored flexibility would allow us to do so 💪